packages/ra-core/src/dataProvider/useUpdate.ts

```diff
@@ -81,87 +82,57 @@ export const useUpdate = <RecordType extends Record = Record>(
     const { mutationMode = 'pessimistic', ...reactMutationOptions } = options;
     const mode = useRef<MutationMode>(mutationMode);
     const paramsRef = useRef<Partial<UpdateParams<RecordType>>>({});
-    const rollbackData = useRef<{
-        previousGetOne?: any;
-        previousGetList?: any;
-        previousGetMany?: any;
-        previousGetManyReference?: any;
-    }>({});
+    const snapshot = useRef<Snapshot>([]);
 
-    const updateCache = async ({ resource, id, data }) => {
+    const updateCache = ({ resource, id, data }) => {
         // hack: only way to tell react-query not to fetch this query for the next 5 seconds
         // because setQueryData doesn't accept a stale time option
-        const updatedAt =
-            mode.current === 'undoable' ? Date.now() + 1000 * 5 : Date.now();
-        await queryClient.setQueryData(
+        const now = Date.now();
+        const updatedAt = mode.current === 'undoable' ? now + 5 * 1000 : now;
+
+        const updateColl = (old: RecordType[]) => {
```

Comment:
Rename `updateColl` to `updateCall`
```diff
-const updateColl = (old: RecordType[]) => {
+const updateCall = (old: RecordType[]) => {

Reply:
its short for "updateCollection"... But maybe it's not clear enough. I preferred the abbreviation to the more explicit "updateCollection" to avoid triggering more line breaks, which harms legibility.

Reply:
Ok, my bad 😄